tests/framework/http_api.py

```diff
@@ -58,15 +58,25 @@ def __init__(self, api, resource, id_field=None):
     def get(self):
         """Make a GET request"""
         url = self._api.endpoint + self.resource
-        res = self._api.session.get(url)
+        try:
+            res = self._api.session.get(url)
+        except Exception as e:
+            if self._api.error_callback:
+                self._api.error_callback("GET", self.resource, str(e))
+            raise
```

Comment:
maybe move `error_callback` into `Api` class and add exception catching there so it will apply to all api calls?